File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.js
function solution(s) {
    let answer = 0;
    
    const sLen = s.length;
    for (let i = 0; i < sLen; i++) {
        const stack = [];
        s.split('').forEach((v) => {
            if (stack.length === 0) stack.push(v);
            else if (v === ")" && stack[stack.length - 1] === "(") stack.pop();
            else if (v === "}" && stack[stack.length - 1] === "{") stack.pop();
            else if (v === "]" && stack[stack.length - 1] === "[") stack.pop();
            else stack.push(v);
        });
        
        if (stack.length === 0) answer++;
        
        s = s.split('').map((v, i, arr) => arr[(i+1) % sLen]).join('');
    }
    return answer;
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    answer = 0
    
    for _ in range(len(s)):
        stack = []
        
        for bracket in s:
            if (len(stack) == 0):
                stack.append(bracket)
            elif bracket == "]" and stack[-1] == "[":
                stack.pop()
            elif bracket == "}" and stack[-1] == "{":
                stack.pop()
            elif bracket == ")" and stack[-1] == "(":
                stack.pop()
            else:
                stack.append(bracket)

        if (len(stack) == 0):
            answer += 1
            
        s = s[1:] + s[0]
    
    return answer